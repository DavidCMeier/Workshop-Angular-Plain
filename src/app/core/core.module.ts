import { StoreModule } from "@ngrx/store";
import { EffectsModule } from "@ngrx/effects";
import * as fromStore from './store';
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HeaderComponent } from './components/header/header.component';
import { FooterComponent } from './components/footer/footer.component';
import { ShellComponent } from './containers/shell/shell.component';
import { MatToolbarModule } from "@angular/material/toolbar";
import { RouterModule } from "@angular/router";
import { MatSelectModule } from "@angular/material/select";
import { MatIconModule } from "@angular/material/icon";
import { MatButtonModule } from "@angular/material/button";
import { HTTP_INTERCEPTORS, HttpClientModule } from "@angular/common/http";
import { AuthInterceptor } from "./interceptors/auth-interceptor.service";
import { SharedModule } from "../shared/shared.module";
import { StoreRouterConnectingModule } from "@ngrx/router-store";



@NgModule({
  declarations: [
    HeaderComponent,
    FooterComponent,
    ShellComponent
  ],
  imports: [
    StoreModule.forFeature('core', fromStore.reducers),
    EffectsModule.forFeature(fromStore.effects),
    StoreRouterConnectingModule.forRoot(),
    CommonModule,
    MatToolbarModule,
    RouterModule,
    MatSelectModule,
    MatIconModule,
    MatButtonModule,
    HttpClientModule,
    SharedModule
  ],
  providers: [
    { provide: HTTP_INTERCEPTORS, useClass: AuthInterceptor, multi: true }
  ],
})
export class CoreModule { }
