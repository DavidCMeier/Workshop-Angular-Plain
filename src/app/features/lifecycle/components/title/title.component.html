  <h1 #titleElement [appSizeText]="size">{{title}}</h1>
