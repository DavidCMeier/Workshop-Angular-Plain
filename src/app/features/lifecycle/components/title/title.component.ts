import {
  AfterViewInit,
  Component,
  ElementRef,
  Input,
  OnChanges,
  OnDestroy,
  OnInit,
  SimpleChanges,
  ViewChild,
  ViewEncapsulation
} from '@angular/core';
import { Size } from "../../../../shared/models/size";

@Component({
  selector: 'app-title',
  templateUrl: './title.component.html',
  styleUrls: ['./title.component.scss'],
  encapsulation: ViewEncapsulation.None
})
export class TitleComponent implements OnInit, OnChanges, AfterViewInit, OnDestroy {

  @Input() title!: string;
  @Input() size!: Size;
  @ViewChild('titleElement') titleElement!: ElementRef;
  constructor() {
    console.log('Constructor', this.title, this.size, this.titleElement);
  }

  ngOnInit(): void {
    console.log('OnInit', this.title, this.size, this.titleElement);
  }

  ngAfterViewInit(): void {
    console.log('AfterViewInit', this.title, this.size, this.titleElement);
  }

  ngOnChanges(changes: SimpleChanges): void {
    console.log('OnChanges',changes);
  }

  ngOnDestroy(): void {
    console.log('OnDestroy');
  }

}
