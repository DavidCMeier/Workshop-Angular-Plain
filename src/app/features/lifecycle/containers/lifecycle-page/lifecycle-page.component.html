<app-title [title]="title" [size]="size"></app-title>
<button (click)="changeTitle()">Cambiar titulo</button>
