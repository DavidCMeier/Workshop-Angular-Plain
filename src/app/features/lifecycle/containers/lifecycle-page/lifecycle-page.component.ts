import { Component, OnInit } from '@angular/core';
import { Size } from "../../../../shared/models/size";

@Component({
  selector: 'app-lifecycle-page',
  templateUrl: './lifecycle-page.component.html',
  styleUrls: ['./lifecycle-page.component.scss']
})
export class LifecyclePageComponent implements OnInit {

  title = 'Ciclo de vida';
  size = Size.LARGE;

  constructor() { }

  ngOnInit(): void {
  }

  changeTitle() {
    this.title = this.title.split("").reverse().join("");
  }
}
