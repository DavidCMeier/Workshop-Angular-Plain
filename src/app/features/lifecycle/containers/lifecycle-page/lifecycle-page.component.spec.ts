import { ComponentFixture, TestBed } from '@angular/core/testing';

import { LifecyclePageComponent } from './lifecycle-page.component';
import { TitleComponent } from "../../components/title/title.component";
import { SizeTextDirective } from "../../../../shared/directives/size-text.directive";

describe('LifecyclePageComponent', () => {
  let component: LifecyclePageComponent;
  let fixture: ComponentFixture<LifecyclePageComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ LifecyclePageComponent, TitleComponent, SizeTextDirective ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(LifecyclePageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it('should change title when click in button', () => {
    const button = fixture.debugElement.nativeElement.querySelector('button');
    button.click();
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot()
  });
  it('should reverse word', () => {
    component.title = 'Hello';
    component.changeTitle();
    const result = 'olleH';
    expect(component.title).toBe(result)
  });
});
