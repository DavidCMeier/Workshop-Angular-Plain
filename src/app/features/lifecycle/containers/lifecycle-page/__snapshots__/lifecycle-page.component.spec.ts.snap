// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LifecyclePageComponent should change title when click in button 1`] = `
<app-lifecycle-page
  __ngContext__={[Function LRootView]}
  size={[Function String]}
  title={[Function String]}
>
  <app-title>
    <h1
      style="font-size: 1.4rem;"
    >
      adiv ed olciC
    </h1>
  </app-title><button>
    Cambiar titulo
  </button>
</app-lifecycle-page>
`;

exports[`LifecyclePageComponent should create 1`] = `
<app-lifecycle-page
  __ngContext__={[Function LRootView]}
  size={[Function String]}
  title={[Function String]}
>
  <app-title>
    <h1
      style="font-size: 1.4rem;"
    >
      Ciclo de vida
    </h1>
  </app-title><button>
    Cambiar titulo
  </button>
</app-lifecycle-page>
`;
